File: test_api.dart
import 'package:http/http.dart' as http;

void main() async {
  final url1 = Uri.parse('https://www.jiosaavn.com/api.php?__call=autocomplete.get&_format=json&_marker=0&api_version=4&ctx=web6dot0&query=taylor');
  final r1 = await http.get(url1, headers: {'User-Agent': 'Mozilla/5.0'});
  print('--- AUTOCOMPLETE ---');
  print(r1.body.substring(0, 100));

  final url2 = Uri.parse('https://www.jiosaavn.com/api.php?__call=search.getResults&_format=json&p=1&n=1&api_version=4&ctx=web6dot0&q=taylor');
  final r2 = await http.get(url2, headers: {'User-Agent': 'Mozilla/5.0'});
  print('--- GETRESULTS ---');
  print(r2.body.substring(0, 100));
}

File: test_api.py
import urllib.request
import json
import urllib.parse

url = 'https://www.jiosaavn.com/api.php?__call=search.getResults&_format=json&p=1&n=20&api_version=4&ctx=web6dot0&q=Arijit%20Singh'
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
try:
    with urllib.request.urlopen(req) as response:
        data = json.loads(response.read().decode())
        if 'results' in data:
            print('results is a List. Length:', len(data['results']))
        else:
            print('keys:', data.keys())
            if 'songs' in data:
                print('songs type:', type(data['songs']))
                if isinstance(data['songs'], dict) and 'data' in data['songs']:
                    print('songs.data length:', len(data['songs']['data']))
except Exception as e:
    print(e)
